jenkins/add-docker-build-job.sh

```diff
@@ -0,0 +1,117 @@
+
+wget ${base_remote_jenkins_scripts}/basic-docker-build-job.xml -O job_template.xml
+wget ${base_remote_jenkins_scripts}/basic-user-pwd-credentials.xml -O credentials_template.xml
+wget ${base_remote_jenkins_groovy}/basic-docker-build.groovy -O job.groovy
+
+#prepare credentials.xml
+cat credentials_template.xml | sed -e "s|{insert-credentials-id}|${cr_credentials_id}|"\
+                                   -e "s|{insert-credentials-description}|${cr_credentials_description}|"\
+                                   -e "s|{insert-user-name}|${cr_user}|"\
+                                   -e "s|{insert-user-password}|${cr_password}|"\
+                                    > credentials.xml
+rm credentials_template.xml
+
+#prepare job.xml
+cat job_template.xml | sed -e "s|{insert-job-display-name}|${job_display_name}|"\
+                           -e "s|{insert-job-description}|${job_description}|"\
+                           -e "s|{insert-git-url}|${git_url}|"\
+                           -e "s|{insert-cr-url}|${cr_url}|"\
+                           -e "s|{insert-docker-credentials}|${cr_credentials_id}|"\
+                           -e "s|{insert-container-tag-prefix}|${container_tag_prefix}|" > job1.xml
+cat job1.xml | sed "/{insert-groovy-script}/r job.groovy" | sed "/{insert-groovy-script}/d" > job.xml
+rm job_template.xml
+rm job1.xml
+rm job.groovy
+set +e
+
+function retry_until_successful {
+    ${@}
+    while [ $? -ne 0 ]; do
+        sleep 5
+        ${@}
+    done;
+}
+
+#download jenkins cli (wait for Jenkins to be online)
+retry_until_successful wget ${jenkins_url}/jnlpJars/jenkins-cli.jar -O jenkins-cli.jar
+
+#install the required plugins
+retry_until_successful java -jar jenkins-cli.jar -s ${jenkins_url} install-plugin "credentials" -deploy --username ${jenkins_user} --password ${jenkins_password}
+retry_until_successful java -jar jenkins-cli.jar -s ${jenkins_url} install-plugin "workflow-cps" -deploy --username ${jenkins_user} --password ${jenkins_password}
+retry_until_successful java -jar jenkins-cli.jar -s ${jenkins_url} install-plugin "workflow-job" -deploy --username ${jenkins_user} --password ${jenkins_password}
+retry_until_successful java -jar jenkins-cli.jar -s ${jenkins_url} install-plugin "docker-workflow" -restart --username ${jenkins_user} --password ${jenkins_password}
+
+#wait for instance to be back online
+retry_until_successful java -jar jenkins-cli.jar -s ${jenkins_url} version --username ${jenkins_user} --password ${jenkins_password}
+
+#add user/pwd
+retry_until_successful cat credentials.xml | java -jar jenkins-cli.jar -s ${jenkins_url} create-credentials-by-xml SystemCredentialsProvider::SystemContextResolver::jenkins "(global)" --username ${jenkins_user} --password ${jenkins_password}
+#add job
+retry_until_successful cat job.xml | java -jar jenkins-cli.jar -s ${jenkins_url} create-job ${job_name} --username ${jenkins_user} --password ${jenkins_password}
+
+#cleanup
+rm credentials.xml
```

Comment:
Are you intentionally leaving 'job.groovy', 'job_template.xml' and 'credentials_template.xml'?